index.d.ts

```diff
@@ -0,0 +1,1353 @@
+
+  /**
+   * Native service object passed from BleModule.
+   * @private
+   */
+  export interface NativeService {
+    /**
+     * Service unique identifier
+     * @private
+     */
+    id: Identifier;
+    /**
+     * Service UUID
+     * @private
+     */
+    uuid: UUID;
+    /**
+     * Device's ID to which service belongs
+     * @private
+     */
+    deviceID: Device['id'];
+    /**
+     * Value indicating whether the type of service is primary or secondary.
+     * @private
+     */
+    isPrimary: boolean;
+  }
+
+  /**
+   * Native characteristic object passed from BleModule.
+   * @private
+   */
+  export interface NativeCharacteristic {
+    /**
+     * Characteristic unique identifier
+     * @private
+     */
+    id: Identifier;
+    /**
+     * Characteristic UUID
+     * @private
+     */
+    uuid: UUID;
+    /**
+     * Service's ID to which characteristic belongs
+     * @private
+     */
+    serviceID: Identifier;
+    /**
+     * Service's UUID to which characteristic belongs
+     * @private
+     */
+    serviceUUID: UUID;
+    /**
+     * Device's ID to which characteristic belongs
+     * @private
+     */
+    deviceID: Device['id'];
+    /**
+     * True if characteristic can be read
+     * @private
+     */
+    isReadable: boolean;
+    /**
+     * True if characteristic can be written with response
+     * @private
+     */
+    isWritableWithResponse: boolean;
+    /**
+     * True if characteristic can be written without response
+     * @private
+     */
+    isWritableWithoutResponse: boolean;
+    /**
+     * True if characteristic can monitor value changes.
+     * @private
+     */
+    isNotifiable: boolean;
+    /**
+     * True if characteristic is monitoring value changes without ACK.
+     * @private
+     */
+    isNotifying: boolean;
+    /**
+     * True if characteristic is monitoring value changes with ACK.
+     * @private
+     */
+    isIndicatable: boolean;
+    /**
+     * Characteristic value if present
+     * @private
+     */
+    value?: Base64;
+  }
+
+  /**
+   * Object representing information about restored BLE state after application relaunch.
+   * @private
+   */
+  export interface NativeBleRestoredState {
+    /**
+     * List of connected devices after state restoration.
+     * @type {Array<NativeDevice>}
+     * @instance
+     * @memberof NativeBleRestoredState
+     * @private
+     */
+    connectedPeripherals: NativeDevice[];
+  }
+
+  /**
+   * Characteristic object.
+   */
+  export class Characteristic implements NativeCharacteristic {
+    /**
+     * Characteristic unique identifier
+     */
+    id: Identifier;
+    /**
+     * Characteristic UUID
+     */
+    uuid: UUID;
+    /**
+     * Service's ID to which characteristic belongs
+     */
+    serviceID: Identifier;
+    /**
+     * Service's UUID to which characteristic belongs
+     */
+    serviceUUID: UUID;
+    /**
+     * Device's ID to which characteristic belongs
+     */
+    deviceID: DeviceId;
```

Comment:
Not declared. Currently we have mixed usage of `Device['id']` and `string`. Maybe it's better to go back to `DeviceId` alias and declare it?